fwprovider/acme/resource_acme_account.go

```
@@ -0,0 +1,258 @@
+	createRequest.Directory = plan.Directory.ValueString()
+	createRequest.EABHMACKey = plan.EABHMACKey.ValueString()
+	createRequest.EABKID = plan.EABKID.ValueString()
+	createRequest.Name = plan.Name.ValueString()
+	createRequest.TOS = plan.TOS.ValueString()
+
+	err := r.client.Create(ctx, createRequest)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Unable to create ACME account '%s'", plan.Name),
+			err.Error(),
+		)
+
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read retrieves the current state of the ACME account from the Proxmox cluster.
+func (r *acmeAccountResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state acmeAccountModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	account, err := r.client.Get(ctx, state.Name.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Unable to read ACME account '%s'", state.Name),
+			err.Error(),
+		)
+
+		return
+	}
+
+	state.Directory = types.StringValue(account.Directory)
+	state.TOS = types.StringValue(account.TOS)
+	// XXX account.Location?
+	// XXX account.Account?
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update modifies an existing ACME account on the Proxmox cluster.
+func (r *acmeAccountResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var plan acmeAccountModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	updateRequest := &account.ACMEAccountUpdateRequestBody{}
+	updateRequest.Contact = plan.Contact.ValueString()
+	updateRequest.Name = plan.Name.ValueString() // XXX name can't be changed?
+
+	err := r.client.Update(ctx, plan.Name.ValueString(), updateRequest)
```

See above, it probably makes sense to use the name from state here instead of plan?

---

After thinking about this for a while, I reverted this change, because I think changing the name should fail instead of silently succeed.